client/payments-welcome/index.tsx

```diff
@@ -0,0 +1,288 @@
+			// 	path: '/wc-calypso-bridge/v1/payments/activate-promo',
+			// 	method: 'POST',
+			// } ) ) as activatePromoResponse;
+
+			// if ( activatePromoRequest?.success ) {
+			window.location.href = connectUrl;
+			// }
+		} catch ( e ) {
+			// renderErrorMessage( e.message );
+		}
+	};
+
+	const handleSetup = async () => {
+		setSubmitted( true );
+		recordEvent( 'wcpay_connect_account_clicked', {
+			// eslint-disable-next-line camelcase
+			wpcom_connection: isJetpackConnected ? 'Yes' : 'No',
+		} );
+
+		try {
+			const installAndActivateResponse = await installAndActivatePlugins(
+				[ 'woocommerce-payments' ]
+			);
+
+			if ( installAndActivateResponse?.success ) {
+				activatePromo();
+			} else {
+				renderErrorMessage( installAndActivateResponse.message );
+			}
+		} catch ( e ) {
+			renderErrorMessage( e.message );
+		}
+	};
+
+	const handleNoThanks = () => {
+		setNoThanksClicked( true );
+		setExitSurveyModalOpen( true );
+	};
+
+	return (
+		<Card className="connect-account__card">
+			<CardHeader>
+				<div>
+					<h1 className="banner-heading-copy">
+						{ strings.bannerHeading }
+					</h1>
+					<TermsOfService />
+				</div>
+				<div className="connect-account__action">
+					<Button
+						isSecondary
+						isBusy={ isNoThanksClicked && isExitSurveyModalOpen }
+						disabled={ isNoThanksClicked && isExitSurveyModalOpen }
+						onClick={ handleNoThanks }
+						className="btn-nothanks"
+					>
+						{ strings.nothanks }
+					</Button>
+					<Button
+						isPrimary
+						isBusy={ isSubmitted }
+						disabled={ isSubmitted }
+						onClick={ handleSetup }
+						className="btn-install"
+					>
+						{ strings.button }
+					</Button>
+					{ isExitSurveyModalOpen && (
+						<ExitSurveyModal
+							setExitSurveyModalOpen={ setExitSurveyModalOpen }
+						/>
+					) }
+				</div>
+			</CardHeader>
+			<CardBody>
+				<div className="content">
+					<p className="onboarding-description">
+						{ strings.onboarding.description }
+						<br />
+						<LearnMore />
+					</p>
+
+					<h3 className="accepted-payment-methods">
+						{ strings.paymentMethodsHeading }
+					</h3>
+
+					<PaymentMethods />
+				</div>
+			</CardBody>
+		</Card>
+	);
+};
+
+const ConnectAccountPage = () => {
+	const [ errorMessage, setErrorMessage ] = useState( '' );
+
+	const { isJetpackConnected, connectUrl, hasViewedWelcomePage } = useSelect(
+		( select ) => {
+			const { getOption } = select( OPTIONS_STORE_NAME );
+			return {
+				isJetpackConnected: select(
+					'wc/admin/plugins'
+				).isJetpackConnected(),
+				connectUrl:
+					'admin.php?wcpay-connect=1&_wpnonce=' +
+					getSetting( 'wcpay_welcome_page_connect_nonce' ),
+				hasViewedWelcomePage:
+					getOption( 'wc_pay_welcome_page_viewed_timestamp' ) ||
+					false,
+			};
+		}
+	);
+	const { updateOptions } = useDispatch( OPTIONS_STORE_NAME );
+
+	/**
+	 * Submits a request to store viewing welcome time.
+	 */
+	const storeViewWelcome = () => {
+		if (
```

Comment:
If we can make sure `hasViewedWelcomePage`  is a boolean value, I think we can simplify this.

```js
if ( !hasViewedWelcomePage ) {
	updateOptions( {
		wc_pay_welcome_page_viewed_timestamp: Math.floor(
			Date.now() / 1000
		),
	} );
}
```

Reply:
Thank you for the suggestion. I left a comment in https://github.com/woocommerce/woocommerce-admin/pull/8083#discussion_r774927238